### PR TITLE
Add nuget packaging support

### PR DESCRIPTION
**Pull request by peppy**
Can't figure how to get the native dlls to only be packaged once via `dotnet pack`; currently they are being included in both `content` and `contentFiles`.

Apart from that, this works well enough.

Todo:
- [x] debug -> release
- [x] versioning
- [x] appveyor_deploy

### CODE REVIEW

**Comment by peppy** [comment]
Ready for review 👍 ⚡️

**Comment by peppy** [comment]
Testable via https://github.com/ppy/osu/pull/2732, but recommended cloning fresh if you value your submodule.

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -175,7 +176,11 @@ protected GameHost(string gameName = @"")
                 (InputThread = new InputThread(null)), //never gets started.
             };
 
-            var path = Path.GetDirectoryName(FullPath);
+            var assembly = Assembly.GetEntryAssembly();
+            if (assembly == null || assembly.Location.Contains("testhost"))
```
Comment (smoogipoo) [inline osu.Framework/Platform/GameHost.cs]: What's this for?

Comment (peppy) [reply osu.Framework/Platform/GameHost.cs]: When running under nUnit the entry assembly becomes nunit, so we need to use the calling assembly instead.

Comment (smoogipoo) [reply osu.Framework/Platform/GameHost.cs]: Needs a comment ^^

**Pull request merged**